src/protobuf/util.ts

```diff
@@ -50,16 +49,56 @@ export function toJSON(tokens: PBToken[]): string {
   );
 }
 
-export function bitob(n: bigint | number): Buffer {
+export function bitob(n: bigint | number): Uint8Array {
   let hv = n.toString(16);
   hv = "".padStart(hv.length % 2, "0") + hv;
-  return Buffer.from(hv, "hex");
+  return hextou8(hv);
 }
 
-export function btobi(buf: Buffer): bigint {
-  return BigInt(`0x${buf.toString("hex")}`);
+export function btobi(buf: Uint8Array): bigint {
+  return BigInt(`0x${u8tohex(buf)}`);
 }
 
-export function cc(args: Buffer[]) {
-  return Buffer.concat(args);
+export function concatu8(args: Uint8Array[]): Uint8Array {
+  let totalLength = 0;
+  for (let i = 0; i < args.length; ++i) {
+    totalLength += args[i].length;
+  }
+  const out = new Uint8Array(totalLength);
+  let offset = 0;
+  for (let i = 0; i < args.length; ++i) {
+    out.set(args[i], offset);
+    offset += args[i].length;
+  }
+  return out;
+}
+
+export function hextou8(data: string): Uint8Array {
+  data =
+    data.startsWith("0x") || data.startsWith("0X") ? data.substring(2) : data;
+  const out = new Uint8Array(data.length / 2);
+  for (let i = 0; i < out.length; ++i) {
+    out[i] = parseInt(data.substr(i * 2, 2), 16);
+  }
+  return out;
 }
+
+export function u8tohex(data: Uint8Array): string {
+  let out = "";
+  for (let i = 0; i < data.length; ++i) {
+    out += data[i].toString(16).padStart(2, "0");
+  }
+  return out;
+}
+
+const _atob = globalThis.atob as ((data: string) => string) | undefined;
+const _btoa = globalThis.btoa as ((data: string) => string) | undefined;
+export const b64tou8 = _atob
```

Comment:
These try to use `atob`/`btoa` in browsers, and fall back to `Buffer` in node.js, where those functions are not defined. `atob`/`btoa` has issues with handling non-ascii text, but this is not a problem for us (as far as I can tell)